File: enchantron-apple/Enchantron Shared/Animation.swift
//
//  Animation.swift
//  Enchantron
//
//  Created by Kevin Guthrie on 5/16/20.
//  Copyright © 2020 Rook And Pawn Industries, Inc. All rights reserved.
//

import Foundation
import Metal
import MetalKit
import simd

class Animation {
    
    var frames: [Texture] = []
    var isLoop = false
    var name = "Un-named animation"
    
    func addTexture(_ texture: Texture) {
        frames.append(texture)
    }
    
    func setIsLoop(_ isLoop: Bool) {
        self.isLoop = isLoop
    }
    
    func setName(_ name: String) {
        self.name = name
    }
}

File: enchantron-core/src/native/animation.rs
use super::Texture;

pub trait Animation: 'static + Send + Sync + Unpin {
    type Texture: Texture;

    fn add_texture(&self, texture: &Self::Texture);

    fn set_is_loop(&self, is_loop: bool);

    fn set_name(&self, name: String);
}
